### DISCUSSION TITLE
[Help]: Fast clicking the next/prev buttons is causing stuttering

### DISCUSSION THREAD

**Discussion by matthias-schaefer** [davidjerleke/embla-carousel#748] [Help] [answered]
### Summary

I'm trying so set up a carousel with pagination buttons and dots, unfortunately, if you quickly click on the next/prev buttons, the animation is not running smoothly, but it begins to stutter. I've attached a link to the codebox example.

This is **not happening** with your official example, that can be found here https://www.embla-carousel.com/examples/predefined/#arrows--dots, so I'm not sure what is causing it, display grid in combination with auto-columns maybe?

A solution would be to block the next/prev button, until the animation is finished, but it's not necessary in your example, so that's a bit confusing for me at the moment. Any idea what is causing this problem in my case?

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### CodeSandbox example

https://codesandbox.io/p/sandbox/pensive-wozniak-g8jx7v-g8jx7v

**Comment by davidjerleke** [comment, marked as answer]
@matthias-schaefer I believe you should read [this guide](https://www.embla-carousel.com/guides/previous-and-next-buttons/#button-placement) in the docs.

Cheers,
David

**Reply by matthias-schaefer** [reply]
Ah, I read it, also included the embla__viewport part, but for some reason I misplaced the navigation elements one level too deep, you are right. Thanks!

**Reply by davidjerleke** [reply]
@matthias-schaefer glad you got it sorted!
